modules/core/src/main/scala/org/aossie/agora/model/Ballot.scala

```diff
@@ -33,7 +33,7 @@ case class ScoreBallot(scores: List[(Candidate, Rational)], override val id: Int
 
 }
 
-case class RankBallot(val ranks: List[(Candidate, Int)], override val id: Int, w: Rational)
+class RankBallot(val ranks: List[(Candidate, Int)], override val id: Int, w: Rational)
```

Comment:
Follow same recommendation as stated for PrefenceBallot here. Add type parameter for Candidate say C and revert to case class.